src/authStrategies/RemoteAuth.js

```diff
@@ -0,0 +1,221 @@
+            }
+            /* Compress & Store Session */
+            const pathExists = await this.isValidPath(this.userDataDir);
+            if (pathExists) {
+                await this.compressSession();
+                await this.store.save({session: this.sessionName});
+                await fs.promises.unlink(`${this.sessionName}.zip`);
+            }
+        } catch (error) {
+            console.log('RemoteAuth Error => storeRemoteSession: ', error);
+        }
+    }
+
+    async extractRemoteSession() {
+        try {
+            const sessionExists = await this.store.sessionExists({session: this.sessionName});
+            const pathExists = await this.isValidPath(this.userDataDir);
+            if (pathExists) {
+                await fs.promises.rm(this.userDataDir, {
+                    recursive: true,
+                    force: true
+                });
+            }
+            if (sessionExists) {
+                await this.store.extract({session: this.sessionName});
+                await this.unCompressSession();
+                await fs.promises.unlink(`RemoteAuth-${this.sessionName}.zip`);
+            }
+        } catch (error) {
+            console.log('RemoteAuth Error => extractRemoteSession: ', error);
+        }
+    }
+
+    async deleteRemoteSession() {
+        try {
+            const sessionExists = await this.store.sessionExists({session: this.sessionName});
+            if (sessionExists) await this.store.delete({session: this.sessionName});
+
+        } catch (error) {
+            console.log('RemoteAuth Error => deleteRemoteSession: ', error);
+        }
+    }
+
+    async compressSession() {
+        try {
+            const archive = archiver('zip');
+            const stream = fs.createWriteStream(`${this.sessionName}.zip`);
+
+            await this.deleteMetadata();
+            return new Promise((resolve, reject) => {
+                archive
+                    .directory(this.userDataDir, false)
+                    .on('error', err => reject(err))
+                    .pipe(stream);
+
+                stream.on('close', () => resolve());
+                archive.finalize();
+            });
+        } catch (error) {
+            console.log('RemoteAuth Error => compressSession: ', error);
+        }
+    }
+
+    async unCompressSession() {
+        try {
+            var stream = fs.createReadStream(`RemoteAuth-${this.sessionName}.zip`);
+            return new Promise((resolve, reject) => {
+                stream.pipe(unzipper.Extract({
+                    path: this.userDataDir
+                }))
+                    .on('error', err => reject(err))
+                    .on('finish', () => resolve());
+            });
+        } catch (error) {
+            console.log('RemoteAuth Error => unCompressSession: ', error);
+        }
+    }
+
+    async deleteMetadata() {
+        try {
+            /* First Level */
```

Comment:
These "two levels" can probably be refactored to share code since the only thing that changes is the path that `readdir` is using, e.g.:

```js
const dirs = [this.userDataDir, path.join(this.userDataDir, "Default")];
for(const metaDir of dirs) {
    const sessionFiles = await fs.promises.readdir(metaDir);
   // ...
}
```

Reply:
Yeah I agree, this needed a refactor. Thanks! 

> Done